modules/Channel.md

```diff
@@ -0,0 +1,31 @@
+## Channel Module
+
+A channel is a FIFO data structure used to communicate between different Chapel tasks.
+Since the operations on the channel are blocking by default, it enables to send values from one task to another in a synchronized manner.
+
+The module will have a `Channel` class to store the channel buffer and support operations on them.
+
+* Here are some functions that the module should have
+
+	*  `send(value : eltType)` : This function waits for a space in the buffer and sends the `value` to the channel.
+
+	*  `recv() : eltType` : This function waits until the buffer is empty and returns the value sent to the channel.
+
+	*  `close()` : This function is used to close the channel such that no more values will be sent to it.
+
+* `iter these()` : An iterator to receive the values sent to the channel until it is closed.
+
+* `select` statements : This statement lets you wait for multiple channel operations and executes if anyone of them is ready for communication.
+
+Some related questions -
+
+* Should this module be standard or package ?
+
+* What should be the name of the module ?
+	 * proposal : `Channels`
```

Comment:
since we already have `channel` for a file object - https://chapel-lang.org/docs/modules/standard/IO.html#IO.channel - having the name of this be `Channel` seems a bit too subtle to me assuming they are both in the same standard library. 

I think it's interesting to consider if a single `channel` type could support both use cases, but I think that's probably a stretch. @Yudhishthira1406 - have you thought about this question?

If they are two different types then we probably need a different term for one or the other.

Reply:
> I think it's interesting to consider if a single `channel` type could support both use cases

Sounds like an interesting approach to me. I think it might make sense if our newly implemented `channels` module somehow also incorporates the functionality of `channel` for a file object.

Reply:
My understanding is that we're thinking that we wouldn't want them to be the same at least because having typed elements is important for Go channels but besides there are other functionality differences.